jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpConnection.java

```diff
@@ -341,9 +342,9 @@ void releaseRequestBuffer()
         {
             if (LOG.isDebugEnabled())
                 LOG.debug("releaseRequestBuffer {}", this);
-            if (_retainableByteBuffer.release())
-                _retainableByteBuffer = null;
-            else
+            RetainableByteBuffer buffer = _retainableByteBuffer;
+            _retainableByteBuffer = null;
+            if (!buffer.release())
                 throw new IllegalStateException("unreleased buffer " + _retainableByteBuffer);
```

Comment:
This will now print `unreleased buffer null`, should use the `buffer` local variable, not the field.